src/Raven.Server/NotificationCenter/NotificationsStorage.cs

```diff
@@ -80,30 +80,32 @@ public void Initialize(StorageEnvironment environment, TransactionContextPool co
 
         public bool Store(Notification notification, DateTime? postponeUntil = null)
         {
-            if (Logger.IsInfoEnabled)
-                Logger.Info($"Saving notification '{notification.Id}'.");
-
             using (_contextPool.AllocateOperationContext(out TransactionOperationContext context))
-            using (var tx = context.OpenWriteTransaction())
             {
-                // if previous notification had postponed until value pass this value to newly saved notification
-                var existing = Get(notification.Id, context, tx);
-
-                if (postponeUntil == null)
+                using (var tx = context.OpenReadTransaction())
                 {
-                    if (existing?.PostponedUntil == DateTime.MaxValue) // postponed until forever
-                        return false;
+                    // if previous notification had postponed until value pass this value to newly saved notification
+                    var existing = Get(notification.Id, context, tx);
+
+                    if (postponeUntil == null)
+                    {
+                        if (existing?.PostponedUntil == DateTime.MaxValue) // postponed until forever
+                            return false;
 
-                    if (existing?.PostponedUntil != null && existing.PostponedUntil.Value > SystemTime.UtcNow)
-                        postponeUntil = existing.PostponedUntil;
+                        if (existing?.PostponedUntil != null && existing.PostponedUntil.Value > SystemTime.UtcNow)
+                            postponeUntil = existing.PostponedUntil;
+                    }
                 }
 
+                if (Logger.IsInfoEnabled)
+                    Logger.Info($"Saving notification '{notification.Id}'.");
+
                 using (var json = context.ReadObject(notification.ToJson(), "notification", BlittableJsonDocumentBuilder.UsageMode.ToDisk))
+                using (var tx = context.OpenWriteTransaction())
                 {
                     Store(context.GetLazyString(notification.Id), notification.CreatedAt, postponeUntil, json, tx);
```

Comment:
Need to do the check here again, you might have had another tx come in the meantime.

Reply:
In any case someone will be overwritten here.. We identify the notification by the Id, which should be unique to the case.
But when 2 notification with the same Id are concurrently stored, I'm not sure if we should care which one will be written.

Reply:
OK